[smartmeter] Binding causes openHAB to crash

I’ve installed openhab 2.4 stable.

When i define a sml reader thing (smartmeter 2.4 binding) my complete openhab crashes and restarts itself. This happens as long as the thing exists (openhab start -> 30 seconds -> openhab restart).
When i delete the smartmeter sml thing everything is fine.

I went back and deleted the new smartmeter binding and installed a jar for openhab 2.3 from marketplace smartmeter binding
-> Everything runs smooth with this.


## Current Behavior

```
2018-12-29 17:33:13.058 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: The source did not signal an event for 30 seconds and has been terminated.. Closing connection and trying again in 2 seconds...; smartmeter:meter:23055963
java.util.concurrent.TimeoutException: The source did not signal an event for 30 seconds and has been terminated.
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2018-12-29 17:33:27.620 [WARN ] [org.apache.felix.fileinstall        ] - \usr\share\openhab2\addons does not exist, please create it.
2018-12-29 17:33:27.642 [WARN ] [org.apache.felix.fileinstall        ] - Root path does not exist: E:\usr\share\openhab2\addons
2018-12-29 17:33:31.799 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2018-12-29 17:33:35.320 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Time zone set to 'Europe/Berlin'.
```

## Steps to Reproduce (for Bugs)
1. Create new Smartmeter thing



## Your Environment
* Version used: 2.4 stable
* Windows 10 1809

@msteigenberger 

[hs_err_pid11596.log](https://github.com/openhab/openhab2-addons/files/2724657/hs_err_pid11596.log)
[hs_err_pid17636.log](https://github.com/openhab/openhab2-addons/files/2724658/hs_err_pid17636.log)


Protocol is SML

`java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)`


`2019-01-03 17:52:56.531 [DEBUG] [martmeter.internal.SmartMeterHandler] - Initializing Smartmeter handler.
2019-01-03 17:52:56.532 [DEBUG] [martmeter.internal.SmartMeterHandler] - config port = COM5
2019-01-03 17:52:56.532 [DEBUG] [artmeter.internal.sml.SmlMeterReader] - Created SmlDevice instance smartmeter:meter:1a4d2bff with serial connector on port COM5
2019-01-03 17:52:56.536 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:53:26.663 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: The source did not signal an event for 30 seconds and has been terminated.. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.concurrent.TimeoutException: The source did not signal an event for 30 seconds and has been terminated.
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:53:28.698 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:53:58.850 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: The source did not signal an event for 30 seconds and has been terminated.. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.concurrent.TimeoutException: The source did not signal an event for 30 seconds and has been terminated.
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:54:14.752 [WARN ] [org.apache.felix.fileinstall        ] - \usr\share\openhab2\addons does not exist, please create it.
2019-01-03 17:54:14.775 [WARN ] [org.apache.felix.fileinstall        ] - Root path does not exist: E:\usr\share\openhab2\addons
2019-01-03 17:54:16.041 [DEBUG] [org.openhab.binding.smartmeter      ] - BundleEvent [unknown:512] - org.openhab.binding.smartmeter
2019-01-03 17:54:16.067 [DEBUG] [org.openhab.binding.smartmeter      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.core.thing.type.ChannelTypeProvider, org.openhab.binding.smartmeter.internal.SmartMeterChannelTypeProvider}={service.id=148, service.bundleid=240, service.scope=bundle, component.name=org.openhab.binding.smartmeter.internal.SmartMeterChannelTypeProvider, component.id=35} - org.openhab.binding.smartmeter
2019-01-03 17:54:16.069 [DEBUG] [org.openhab.binding.smartmeter      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.config.core.ConfigOptionProvider}={service.id=149, service.bundleid=240, service.scope=bundle, component.name=org.openhab.binding.smartmeter.internal.SmartMeterConfigProvider, component.id=36} - org.openhab.binding.smartmeter
2019-01-03 17:54:16.941 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2019-01-03 17:54:16.955 [DEBUG] [org.openhab.binding.smartmeter      ] - BundleEvent STARTING - org.openhab.binding.smartmeter
2019-01-03 17:54:16.956 [DEBUG] [org.openhab.binding.smartmeter      ] - BundleEvent STARTED - org.openhab.binding.smartmeter
2019-01-03 17:54:19.095 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Time zone set to 'Europe/Berlin'.
2019-01-03 17:54:19.098 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Location set to '50.699494942428345,7.836035181335028'.
2019-01-03 17:54:19.099 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Locale set to 'de_DE'.
2019-01-03 17:54:20.711 [INFO ] [io.openhabcloud.internal.CloudClient] - Connected to the openHAB Cloud service (UUID = 355bd7c0-9e7c-4be9-bc17-3a9c713e116f, base URL = http://localhost:8080)
2019-01-03 17:54:24.928 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'energy.items'
2019-01-03 17:54:25.002 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'graph.items'
2019-01-03 17:54:25.047 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'heating.items'
2019-01-03 17:54:25.108 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'light.items'
2019-01-03 17:54:25.140 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'persistence.items'
2019-01-03 17:54:25.170 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'rollershutter.items'
2019-01-03 17:54:25.196 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'switch.items'
2019-01-03 17:54:25.215 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'temperature.items'
2019-01-03 17:54:25.236 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'time.items'
2019-01-03 17:54:25.248 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'water.items'
2019-01-03 17:54:25.475 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'jdbc.persist'
2019-01-03 17:54:26.417 [INFO ] [thome.model.lsp.internal.ModelServer] - Started Language Server Protocol (LSP) service on port 5007
2019-01-03 17:54:28.918 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'temperature.rules'
2019-01-03 17:54:29.019 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'weihnachten.rules'
2019-01-03 17:54:29.203 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'switch.rules'
2019-01-03 17:54:29.455 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'heating.rules'
2019-01-03 17:54:29.644 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'water.rules'
2019-01-03 17:54:29.883 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'energy.rules'
2019-01-03 17:54:30.257 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'rollershutter.rules'
2019-01-03 17:54:30.564 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'zuhause.sitemap'
2019-01-03 17:54:30.921 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'astro:sun:local' to inbox.
2019-01-03 17:54:30.931 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'astro:moon:local' to inbox.
2019-01-03 17:54:30.940 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:0AB144' to inbox.
2019-01-03 17:54:30.950 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:0954AA' to inbox.
2019-01-03 17:54:30.970 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:0A1B0E' to inbox.
2019-01-03 17:54:31.003 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:0A0FDA' to inbox.
2019-01-03 17:54:31.017 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:0A69AD' to inbox.
2019-01-03 17:54:31.033 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:090D57' to inbox.
2019-01-03 17:54:31.048 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'tplinksmarthome:hs110:0AF0FB' to inbox.
2019-01-03 17:54:31.060 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'harmonyhub:hub:Wohnzimmer' to inbox.
2019-01-03 17:54:31.075 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'chromecast:chromecast:dae23aad571c681fd37c5591d6e54ea8' to inbox.
2019-01-03 17:54:31.087 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'chromecast:chromecast:6f7f48f56356ccd2e8d0d1260124db6e' to inbox.
2019-01-03 17:54:31.096 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'chromecast:chromecast:0c7615480abfb87fa6bae18c41013657' to inbox.
2019-01-03 17:54:37.940 [WARN ] [erreader.internal.MeterReaderHandler] - OBIS 1-0:1.8.0 is not available in SML Heizung!
2019-01-03 17:54:37.940 [WARN ] [erreader.internal.MeterReaderHandler] - OBIS 1-0:16.7.0 is not available in SML Heizung!
2019-01-03 17:54:39.094 [INFO ] [thome.binding.astro.internal.job.Job] - Scheduled Astro event-jobs for thing astro:moon:local
2019-01-03 17:54:39.200 [INFO ] [ing.meterreader.internal.MeterDevice] - Read out following values: Device: meterreader:meter:962361ab
Obis: 1-0:1.8.1 MeterValue [obis=1-0:1.8.1, value=1.6720518600000001E7, unit=WATT_HOUR]
Obis: 1-0:16.7.0 MeterValue [obis=1-0:16.7.0, value=2042.1000000000001, unit=WATT]
Obis: 1-0:1.8.2 MeterValue [obis=1-0:1.8.2, value=0.0, unit=WATT_HOUR]
Obis: 129-129:199.130.5 MeterValue [obis=129-129:199.130.5, value=AA A7 CC 39 96 0E 76 F1 BC 39 BF 20 38 5C 70 6B 75 C1 67 6C F8 93 5A EE 9C C4 7A 6F D4 E8 DB 94 77 09 92 A7 BA B5 28 16 8B DC A3 AA 00 68 F4 C6, unit=EMPTY]
Obis: 129-129:199.130.3 MeterValue [obis=129-129:199.130.3, value=45 4D 48, unit=EMPTY]
Obis: 1-0:0.0.9 MeterValue [obis=1-0:0.0.9, value=09 01 45 4D 48 00 00 53 89 E7, unit=EMPTY]
Obis: 1-0:1.8.0 MeterValue [obis=1-0:1.8.0, value=1.6720518600000001E7, unit=WATT_HOUR]

2019-01-03 17:54:39.546 [INFO ] [thome.binding.astro.internal.job.Job] - Scheduled Astro event-jobs for thing astro:sun:local
2019-01-03 17:54:40.034 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'WIRED' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:54:41.037 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'HMIP' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:54:42.040 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'CUXD' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:54:43.046 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'GROUP' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:54:43.090 [INFO ] [ommunicator.AbstractHomematicGateway] - HmGatewayInfo[id=HOMEGEAR,type=Homegear,firmware=0.8.0-2309,address=VBC2688033,rf=true,wired=false,hmip=false,cuxd=false,group=false]
2019-01-03 17:54:43.317 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'Connect.php' contains invalid characters, new Datapoint name 'Connect_php'
2019-01-03 17:54:43.318 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'RestoreHomegear.sh' contains invalid characters, new Datapoint name 'RestoreHomegear_sh'
2019-01-03 17:54:43.318 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'BackupHomegear.sh' contains invalid characters, new Datapoint name 'BackupHomegear_sh'
2019-01-03 17:54:43.318 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'ReadMe.txt' contains invalid characters, new Datapoint name 'ReadMe_txt'
2019-01-03 17:54:43.318 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'Test.php' contains invalid characters, new Datapoint name 'Test_php'
2019-01-03 17:54:45.693 [WARN ] [core.thing.internal.ThingManagerImpl] - Initializing handler for thing 'chromecast:audio:b3659ce1632cfe0c73fda10fc42131ab' takes more than 5000ms.
2019-01-03 17:54:45.841 [WARN ] [e.jetty.util.thread.ThreadPoolBudget] - Low configured threads: (max=8 - required=1)=7 < warnAt=8 for QueuedThreadPool[ServletModel-12]@509af831{STARTING,8<=0<=8,i=0,q=0}[ReservedThreadExecutor@5cc52ddb{s=0/1,p=0}]
2019-01-03 17:54:46.336 [INFO ] [.dashboard.internal.DashboardService] - Started Dashboard at http://10.0.0.250:8080
2019-01-03 17:54:46.598 [INFO ] [.dashboard.internal.DashboardService] - Started Dashboard at https://10.0.0.250:8443
2019-01-03 17:54:46.725 [DEBUG] [martmeter.internal.SmartMeterHandler] - Initializing Smartmeter handler.
2019-01-03 17:54:46.726 [DEBUG] [martmeter.internal.SmartMeterHandler] - config port = COM5
2019-01-03 17:54:46.743 [DEBUG] [artmeter.internal.sml.SmlMeterReader] - Created SmlDevice instance smartmeter:meter:1a4d2bff with serial connector on port COM5
2019-01-03 17:54:46.904 [DEBUG] [org.openhab.binding.smartmeter      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory}={service.id=367, service.bundleid=240, service.scope=bundle, component.name=org.openhab.binding.smartmeter.internal.SmartMeterHandlerFactory, component.id=37} - org.openhab.binding.smartmeter
2019-01-03 17:54:46.914 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:54:46.928 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: null. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.ConcurrentModificationException: null
	at java.util.HashMap$KeySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at org.eclipse.smarthome.io.transport.serial.internal.SerialPortRegistry.getPortProvidersForPortName(SerialPortRegistry.java:87) ~[?:?]
	at org.eclipse.smarthome.io.transport.serial.internal.SerialPortManagerImpl.getIdentifier(SerialPortManagerImpl.java:85) ~[?:?]
	at org.openhab.binding.smartmeter.internal.sml.SmlSerialConnector.openConnection(SmlSerialConnector.java:100) ~[240:org.openhab.binding.smartmeter:2.4.0]
	at org.openhab.binding.smartmeter.internal.MeterDevice.lambda$0(MeterDevice.java:159) ~[240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle$SubscriptionLambdaSubscriber.onSubscribe(FlowableDoOnLifecycle.java:63) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed.subscribeActual(FlowableTimeoutTimed.java:47) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle.subscribeActual(FlowableDoOnLifecycle.java:38) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnEach.subscribeActual(FlowableDoOnEach.java:50) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle.subscribeActual(FlowableDoOnLifecycle.java:38) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnEach.subscribeActual(FlowableDoOnEach.java:50) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowablePublish.connect(FlowablePublish.java:130) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableRefCount.subscribeActual(FlowableRefCount.java:91) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14426) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableRepeatWhen$WhenReceiver.onNext(FlowableRepeatWhen.java:100) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableRetryWhen.subscribeActual(FlowableRetryWhen.java:62) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14426) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.run(FlowableSubscribeOn.java:82) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:54:47.153 [INFO ] [arthome.ui.paper.internal.PaperUIApp] - Started Paper UI at /paperui
2019-01-03 17:54:48.963 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:54:49.335 [INFO ] [eemulation.internal.ConfigManagement] - Hue Emulation pairing disabled. Service available under /api
2019-01-03 17:54:50.022 [INFO ] [ui.habmin.internal.servlet.HABminApp] - Started HABmin servlet at /habmin
2019-01-03 17:54:50.116 [INFO ] [persistence.jdbc.internal.JdbcMapper] - JDBC::openConnection: Driver is available::Yank setupDataSource
2019-01-03 17:54:50.174 [WARN ] [.zaxxer.hikari.util.DriverDataSource] - Registered driver with driverClassName=com.mysql.jdbc.Driver was not found, trying direct instantiation.
2019-01-03 17:55:19.175 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: The source did not signal an event for 30 seconds and has been terminated.. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.concurrent.TimeoutException: The source did not signal an event for 30 seconds and has been terminated.
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:55:21.218 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:55:51.256 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: The source did not signal an event for 30 seconds and has been terminated.. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.concurrent.TimeoutException: The source did not signal an event for 30 seconds and has been terminated.
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:55:53.301 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:56:23.338 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: The source did not signal an event for 30 seconds and has been terminated.. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.concurrent.TimeoutException: The source did not signal an event for 30 seconds and has been terminated.
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutSubscriber.onTimeout(FlowableTimeoutTimed.java:139) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed$TimeoutTask.run(FlowableTimeoutTimed.java:170) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:56:31.626 [WARN ] [org.apache.felix.fileinstall        ] - \usr\share\openhab2\addons does not exist, please create it.
2019-01-03 17:56:31.636 [WARN ] [org.apache.felix.fileinstall        ] - Root path does not exist: E:\usr\share\openhab2\addons
2019-01-03 17:56:35.195 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2019-01-03 17:56:35.235 [DEBUG] [org.openhab.binding.smartmeter      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.core.thing.type.ChannelTypeProvider, org.openhab.binding.smartmeter.internal.SmartMeterChannelTypeProvider}={service.id=150, service.bundleid=240, service.scope=bundle, component.name=org.openhab.binding.smartmeter.internal.SmartMeterChannelTypeProvider, component.id=35} - org.openhab.binding.smartmeter
2019-01-03 17:56:35.236 [DEBUG] [org.openhab.binding.smartmeter      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.config.core.ConfigOptionProvider}={service.id=151, service.bundleid=240, service.scope=bundle, component.name=org.openhab.binding.smartmeter.internal.SmartMeterConfigProvider, component.id=36} - org.openhab.binding.smartmeter
2019-01-03 17:56:35.237 [DEBUG] [org.openhab.binding.smartmeter      ] - BundleEvent [unknown:512] - org.openhab.binding.smartmeter
2019-01-03 17:56:35.389 [DEBUG] [org.openhab.binding.smartmeter      ] - BundleEvent STARTING - org.openhab.binding.smartmeter
2019-01-03 17:56:35.390 [DEBUG] [org.openhab.binding.smartmeter      ] - BundleEvent STARTED - org.openhab.binding.smartmeter
2019-01-03 17:56:37.411 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Time zone set to 'Europe/Berlin'.
2019-01-03 17:56:37.414 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Location set to '50.699494942428345,7.836035181335028'.
2019-01-03 17:56:37.414 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Locale set to 'de_DE'.
2019-01-03 17:56:38.056 [INFO ] [io.openhabcloud.internal.CloudClient] - Connected to the openHAB Cloud service (UUID = 355bd7c0-9e7c-4be9-bc17-3a9c713e116f, base URL = http://localhost:8080)
2019-01-03 17:56:43.118 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'energy.items'
2019-01-03 17:56:43.188 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'graph.items'
2019-01-03 17:56:43.222 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'heating.items'
2019-01-03 17:56:43.267 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'light.items'
2019-01-03 17:56:43.288 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'persistence.items'
2019-01-03 17:56:43.330 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'rollershutter.items'
2019-01-03 17:56:43.349 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'switch.items'
2019-01-03 17:56:43.372 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'temperature.items'
2019-01-03 17:56:43.390 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'time.items'
2019-01-03 17:56:43.409 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'water.items'
2019-01-03 17:56:43.855 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'jdbc.persist'
2019-01-03 17:56:44.818 [INFO ] [thome.model.lsp.internal.ModelServer] - Started Language Server Protocol (LSP) service on port 5007
2019-01-03 17:56:47.333 [INFO ] [el.core.internal.ModelRepositoryImpl] - Validation issues found in configuration model 'temperature.rules', using it anyway:
The value of the local variable df_day is not used
2019-01-03 17:56:47.335 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'temperature.rules'
2019-01-03 17:56:47.430 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'weihnachten.rules'
2019-01-03 17:56:47.592 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'switch.rules'
2019-01-03 17:56:47.751 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'heating.rules'
2019-01-03 17:56:47.884 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'water.rules'
2019-01-03 17:56:48.078 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'energy.rules'
2019-01-03 17:56:48.375 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'rollershutter.rules'
2019-01-03 17:56:48.639 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'zuhause.sitemap'
2019-01-03 17:56:55.514 [WARN ] [erreader.internal.MeterReaderHandler] - OBIS 1-0:1.8.0 is not available in SML Heizung!
2019-01-03 17:56:55.514 [WARN ] [erreader.internal.MeterReaderHandler] - OBIS 1-0:16.7.0 is not available in SML Heizung!
2019-01-03 17:56:56.488 [INFO ] [thome.binding.astro.internal.job.Job] - Scheduled Astro event-jobs for thing astro:moon:local
2019-01-03 17:56:56.690 [INFO ] [thome.binding.astro.internal.job.Job] - Scheduled Astro event-jobs for thing astro:sun:local
2019-01-03 17:56:57.440 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'WIRED' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:56:57.586 [INFO ] [ing.meterreader.internal.MeterDevice] - Read out following values: Device: meterreader:meter:962361ab
Obis: 1-0:1.8.1 MeterValue [obis=1-0:1.8.1, value=1.67205975E7, unit=WATT_HOUR]
Obis: 1-0:16.7.0 MeterValue [obis=1-0:16.7.0, value=2033.5, unit=WATT]
Obis: 1-0:1.8.2 MeterValue [obis=1-0:1.8.2, value=0.0, unit=WATT_HOUR]
Obis: 129-129:199.130.5 MeterValue [obis=129-129:199.130.5, value=AA A7 CC 39 96 0E 76 F1 BC 39 BF 20 38 5C 70 6B 75 C1 67 6C F8 93 5A EE 9C C4 7A 6F D4 E8 DB 94 77 09 92 A7 BA B5 28 16 8B DC A3 AA 00 68 F4 C6, unit=EMPTY]
Obis: 129-129:199.130.3 MeterValue [obis=129-129:199.130.3, value=45 4D 48, unit=EMPTY]
Obis: 1-0:0.0.9 MeterValue [obis=1-0:0.0.9, value=09 01 45 4D 48 00 00 53 89 E7, unit=EMPTY]
Obis: 1-0:1.8.0 MeterValue [obis=1-0:1.8.0, value=1.67205975E7, unit=WATT_HOUR]

2019-01-03 17:56:57.664 [WARN ] [su.litvak.chromecast.api.v2.Channel ] - Error while reading
su.litvak.chromecast.api.v2.ChromeCastException: Remote socket closed
	at su.litvak.chromecast.api.v2.Channel.read(Channel.java:425) ~[232:org.openhab.binding.chromecast:2.4.0]
	at su.litvak.chromecast.api.v2.Channel.access$200(Channel.java:51) ~[232:org.openhab.binding.chromecast:2.4.0]
	at su.litvak.chromecast.api.v2.Channel$ReadThread.run(Channel.java:137) [232:org.openhab.binding.chromecast:2.4.0]
2019-01-03 17:56:57.675 [WARN ] [su.litvak.chromecast.api.v2.Channel ] -  <--  null payload in message 
2019-01-03 17:56:58.445 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'HMIP' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:56:59.449 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'CUXD' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:57:00.410 [WARN ] [nce.extensions.PersistenceExtensions] - There is no queryable persistence service registered with the id 'jdbc'
2019-01-03 17:57:00.412 [WARN ] [nce.extensions.PersistenceExtensions] - There is no queryable persistence service registered with the id 'jdbc'
2019-01-03 17:57:00.412 [ERROR] [ntime.internal.engine.ExecuteRuleJob] - Error during the execution of rule 'Energy': The argument 'state' must not be null.
2019-01-03 17:57:00.412 [ERROR] [ntime.internal.engine.ExecuteRuleJob] - Error during the execution of rule 'Heizung': The argument 'state' must not be null.
2019-01-03 17:57:00.452 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'GROUP' on gateway '51D140F6-BDA2-EA3E-ACE3-97D6CBC77848' not available, disabling support
2019-01-03 17:57:00.485 [INFO ] [ommunicator.AbstractHomematicGateway] - HmGatewayInfo[id=HOMEGEAR,type=Homegear,firmware=0.8.0-2309,address=VBC2688033,rf=true,wired=false,hmip=false,cuxd=false,group=false]
2019-01-03 17:57:00.760 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'Connect.php' contains invalid characters, new Datapoint name 'Connect_php'
2019-01-03 17:57:00.761 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'RestoreHomegear.sh' contains invalid characters, new Datapoint name 'RestoreHomegear_sh'
2019-01-03 17:57:00.761 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'BackupHomegear.sh' contains invalid characters, new Datapoint name 'BackupHomegear_sh'
2019-01-03 17:57:00.761 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'ReadMe.txt' contains invalid characters, new Datapoint name 'ReadMe_txt'
2019-01-03 17:57:00.761 [INFO ] [ng.homematic.internal.misc.MiscUtils] - Datapoint name 'Test.php' contains invalid characters, new Datapoint name 'Test_php'
2019-01-03 17:57:01.370 [WARN ] [nce.extensions.PersistenceExtensions] - There is no queryable persistence service registered with the id 'jdbc'
2019-01-03 17:57:01.371 [ERROR] [ntime.internal.engine.ExecuteRuleJob] - Error during the execution of rule 'Temperatures': cannot invoke method public abstract java.util.Date org.eclipse.smarthome.core.persistence.HistoricItem.getTimestamp() on null
2019-01-03 17:57:02.767 [WARN ] [core.thing.internal.ThingManagerImpl] - Initializing handler for thing 'chromecast:audio:b3659ce1632cfe0c73fda10fc42131ab' takes more than 5000ms.
2019-01-03 17:57:02.924 [WARN ] [e.jetty.util.thread.ThreadPoolBudget] - Low configured threads: (max=8 - required=1)=7 < warnAt=8 for QueuedThreadPool[ServletModel-12]@6e2d44b6{STARTING,8<=0<=8,i=0,q=0}[ReservedThreadExecutor@abd7e3d{s=0/1,p=0}]
2019-01-03 17:57:03.342 [INFO ] [.dashboard.internal.DashboardService] - Started Dashboard at http://10.0.0.250:8080
2019-01-03 17:57:03.605 [INFO ] [.dashboard.internal.DashboardService] - Started Dashboard at https://10.0.0.250:8443
2019-01-03 17:57:03.693 [DEBUG] [org.openhab.binding.smartmeter      ] - ServiceEvent REGISTERED - {org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory}={service.id=361, service.bundleid=240, service.scope=bundle, component.name=org.openhab.binding.smartmeter.internal.SmartMeterHandlerFactory, component.id=37} - org.openhab.binding.smartmeter
2019-01-03 17:57:03.701 [DEBUG] [martmeter.internal.SmartMeterHandler] - Initializing Smartmeter handler.
2019-01-03 17:57:03.701 [DEBUG] [martmeter.internal.SmartMeterHandler] - config port = COM5
2019-01-03 17:57:03.708 [DEBUG] [artmeter.internal.sml.SmlMeterReader] - Created SmlDevice instance smartmeter:meter:1a4d2bff with serial connector on port COM5
2019-01-03 17:57:03.803 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:57:03.813 [WARN ] [ding.smartmeter.internal.MeterDevice] - Failed to read: null. Closing connection and trying again in 2 seconds...; smartmeter:meter:1a4d2bff
java.util.ConcurrentModificationException: null
	at java.util.HashMap$KeySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at org.eclipse.smarthome.io.transport.serial.internal.SerialPortRegistry.getPortProvidersForPortName(SerialPortRegistry.java:87) ~[?:?]
	at org.eclipse.smarthome.io.transport.serial.internal.SerialPortManagerImpl.getIdentifier(SerialPortManagerImpl.java:85) ~[?:?]
	at org.openhab.binding.smartmeter.internal.sml.SmlSerialConnector.openConnection(SmlSerialConnector.java:100) ~[240:org.openhab.binding.smartmeter:2.4.0]
	at org.openhab.binding.smartmeter.internal.MeterDevice.lambda$0(MeterDevice.java:159) ~[240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle$SubscriptionLambdaSubscriber.onSubscribe(FlowableDoOnLifecycle.java:63) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableTimeoutTimed.subscribeActual(FlowableTimeoutTimed.java:47) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle.subscribeActual(FlowableDoOnLifecycle.java:38) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnEach.subscribeActual(FlowableDoOnEach.java:50) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnLifecycle.subscribeActual(FlowableDoOnLifecycle.java:38) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableDoOnEach.subscribeActual(FlowableDoOnEach.java:50) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowablePublish.connect(FlowablePublish.java:130) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableRefCount.subscribeActual(FlowableRefCount.java:91) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14426) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableRepeatWhen$WhenReceiver.onNext(FlowableRepeatWhen.java:100) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableRetryWhen.subscribeActual(FlowableRetryWhen.java:62) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14479) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.Flowable.subscribe(Flowable.java:14426) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.run(FlowableSubscribeOn.java:82) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) [240:org.openhab.binding.smartmeter:2.4.0]
	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) [240:org.openhab.binding.smartmeter:2.4.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2019-01-03 17:57:03.948 [INFO ] [arthome.ui.paper.internal.PaperUIApp] - Started Paper UI at /paperui
2019-01-03 17:57:05.820 [INFO ] [ding.smartmeter.internal.MeterDevice] - Opening connection to smartmeter:meter:1a4d2bff
2019-01-03 17:57:06.360 [INFO ] [eemulation.internal.ConfigManagement] - Hue Emulation pairing disabled. Service available under /api
2019-01-03 17:57:07.192 [INFO ] [ui.habmin.internal.servlet.HABminApp] - Started HABmin servlet at /habmin
2019-01-03 17:57:07.298 [INFO ] [persistence.jdbc.internal.JdbcMapper] - JDBC::openConnection: Driver is available::Yank setupDataSource`

--- a/addons/binding/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/sml/SmlSerialConnector.java
+++ b/addons/binding/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/sml/SmlSerialConnector.java
@@ -8,6 +8,8 @@
  */
 package org.openhab.binding.smartmeter.internal.sml;
 
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
@@ -85,6 +87,7 @@ public final class SmlSerialConnector extends ConnectorBase<SmlFile> {
         // read out the whole buffer. We are only interested in the most recent SML file.
         Stack<SmlFile> smlFiles = new Stack<>();
         do {
+            logger.trace("Reading {}. SML message", smlFiles.size() + 1);
             smlFiles.push(TRANSPORT.getSMLFile(is));
         } while (is != null && is.available() > 0);
         if (smlFiles.isEmpty()) {
@@ -124,8 +127,8 @@ public final class SmlSerialConnector extends ConnectorBase<SmlFile> {
             }
             // serialPort.setFlowControlMode(SerialPort.FLOWCONTROL_RTSCTS_IN | SerialPort.FLOWCONTROL_RTSCTS_OUT);
             serialPort.notifyOnDataAvailable(true);
-            is = new DataInputStream(serialPort.getInputStream());
-            os = new DataOutputStream(serialPort.getOutputStream());
+            is = new DataInputStream(new BufferedInputStream(serialPort.getInputStream()));
+            os = new DataOutputStream(new BufferedOutputStream(serialPort.getOutputStream()));
         } else {
             throw new IllegalStateException(MessageFormat.format("No provider for port {0} found", getPortName()));
         }

--- a/addons/binding/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/sml/SmlSerialConnector.java
+++ b/addons/binding/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/sml/SmlSerialConnector.java
@@ -111,6 +111,12 @@ public final class SmlSerialConnector extends ConnectorBase<SmlFile> {
             try {
                 serialPort.setSerialPortParams(baudrateToUse, serialParameter.getDatabits(),
                         serialParameter.getStopbits(), serialParameter.getParity());
+                serialPort.setFlowControlMode(SerialPort.FLOWCONTROL_RTSCTS_IN | SerialPort.FLOWCONTROL_RTSCTS_OUT);
+                try {
+                    serialPort.enableReceiveTimeout(100);
+                } catch (UnsupportedCommOperationException e) {
+                    // doesn't matter (rfc2217 is not supporting this)
+                }
             } catch (UnsupportedCommOperationException e) {
                 throw new IOException(MessageFormat.format(
                         "Error at SerialConnector.openConnection: unable to set serial port parameters for port {0}.",

